lib/mongodbext.js

```diff
@@ -157,20 +179,24 @@ Collection.prototype.insertOne = function(doc, options, callback) {
 	};
 	this.trigger('beforeInsertOne', [hookParams], function(err) {
 		if (err) {
-			return callback(err);
+			return triggerErrorHook(err, callback);
 		}
 
 		var sourceInsertCallback = function(err, insertResult) {
 			if (err) {
-				return callback(err);
+				return triggerErrorHook(err, callback);
 			}
 
 			var afterTriggerParams = [{
 				obj: insertResult.ops[0],
 				options: options
 			}];
 			self.trigger('afterInsertOne', afterTriggerParams, function(err) {
-				callback(err, isReturnDocsOnly ? insertResult.ops[0] : insertResult);
+				if (err) {
+					return triggerErrorHook(err, callback);
+				}
+
+				callback(null, isReturnDocsOnly ? insertResult.ops[0] : insertResult);
```

Comment:
This callback will be called only in case of success. If error occurred callback will be called after error hooks with error as first argument.